packages/evm/src/evmTransfer.ts

```diff
@@ -0,0 +1,56 @@
+import type { BaseTransferParams, Config, Eip1193Provider } from '@buildwithsygma/core';
+import { BaseTransfer } from '@buildwithsygma/core';
+import { providers } from 'ethers';
+
+import { TwapFeeCalculator } from './fee/TwapFee.js';
+import { getFeeInformation, BasicFeeCalculator, PercentageFeeCalculator } from './fee/index.js';
+import type { EvmFee } from './types.js';
+
+export interface EvmTransferParams extends BaseTransferParams {
+  sourceNetworkProvider: Eip1193Provider;
+}
+
+export class EvmTransfer extends BaseTransfer {
+  sourceNetworkProvider: Eip1193Provider;
+
+  constructor(params: EvmTransferParams, config: Config) {
+    super(params, config);
+    this.sourceNetworkProvider = params.sourceNetworkProvider;
+  }
+
+  protected getDepositData(): string {
+    return '';
```

Comment:
Probably should throw method is not implemented

Reply:
Or we can remove it completely and implement so far only inside EvmFungibleTransferRequest

Reply:
IMO throwing the method not implemented error is good as `this method is also implemented in generic transfers